### PR TITLE
Sgd refactoring - Added new tracker and debugging info

### PR DESCRIPTION
**Pull request by VahidooX**

### CODE REVIEW

File: examples/nlp/dialogue_state_tracking/dialogue_state_tracking_sgd.py [hunk 1]
```diff
@@ -153,7 +150,19 @@
 )
 
 parser.add_argument(
-    "--enable_pin_memory", action="store_true", help="Enabled the pin_memory feature of Pytroch's DataLoader",
+    "--enable_pin_memory", action="store_true", help="Enables the pin_memory feature of Pytroch's DataLoader",
+)
+
+parser.add_argument(
+    "--state_tracker", type=str, default='baseline', help="Specifies the state tracker mode",
+)
```
Comment (ekmb) [inline examples/nlp/dialogue_state_tracking/dialogue_state_tracking_sgd.py:158]: let's add supported choices here, I'll add mine, could you add options you're using

File: examples/nlp/dialogue_state_tracking/dialogue_state_tracking_sgd.py [hunk 2]
```diff
@@ -153,7 +150,19 @@
...
+
+parser.add_argument(
+    "--log_data_warnings", action="store_true", help="Enables logging warnings for data",
+)
+
+parser.add_argument(
+    "--eval_debug", action="store_true", help="Enables logging evaluation info for debugging",
```
Comment (ekmb) [inline examples/nlp/dialogue_state_tracking/dialogue_state_tracking_sgd.py:165]: can we just use logging.debug for both --log_data__warning and --eval_debug?

File: examples/nlp/dialogue_state_tracking/dialogue_state_tracking_sgd.py [hunk 3]
```diff
@@ -211,7 +220,7 @@
     max_seq_length=args.max_seq_length,
     datasets=['train', args.eval_dataset],
     overwrite_dial_files=args.overwrite_dial_files,
-    log_data_warnings=False,
+    log_data_warnings=args.log_data_warnings,
 )
```
Comment (ekmb) [inline examples/nlp/dialogue_state_tracking/dialogue_state_tracking_sgd.py:224]: it's a leftover from the original code - could we remove it from here and add logging.debug() to the dataprocessor

File: nemo/collections/nlp/data/datasets/sgd_dataset/prediction_utils.py [hunk 4]
```diff
@@ -24,10 +24,161 @@
...
 
-__all__ = ['get_predicted_dialog', 'write_predictions_to_file']
+__all__ = ['get_predicted_dialog_baseline', 'write_predictions_to_file']
 
 
-def get_predicted_dialog(dialog, all_predictions, schemas):
+def get_predicted_dialog_ret_sys_act(dialog, all_predictions, schemas, eval_debug):
+    """Update labels in a dialogue based on model predictions.
+  Args:
+    dialog: A json object containing dialogue whose labels are to be updated.
```
Comment (ekmb) [inline nemo/collections/nlp/data/datasets/sgd_dataset/prediction_utils.py:33]: could you please add comments about the retrieving slots from the system utterance

File: nemo/collections/nlp/data/datasets/sgd_dataset/prediction_utils.py [hunk 5]
```diff
@@ -24,10 +24,161 @@
...
+                    if slot_status == data_utils.STATUS_DONTCARE:
+                        slot_values[slot] = data_utils.STR_DONTCARE
+                    elif slot_status == data_utils.STATUS_ACTIVE:
+                        tok_start_idx = predictions["noncat_slot_start"][slot_idx]
+                        tok_end_idx = predictions["noncat_slot_end"][slot_idx]
+                        ch_start_idx = predictions["noncat_alignment_start"][tok_start_idx]
+                        ch_end_idx = predictions["noncat_alignment_end"][tok_end_idx]
+                        # print(ch_start_idx, ch_end_idx)
+                        # print(f'Active Slot: {slot}')
+                        # print(f'{predictions["noncat_slot_p"][slot_idx]}, ({ch_start_idx}, {ch_end_idx}), {user_utterance[ch_start_idx - 1 : ch_end_idx]}')
```
Comment (ekmb) [inline nemo/collections/nlp/data/datasets/sgd_dataset/prediction_utils.py:153]: could you remove commented out lines?

File: examples/nlp/dialogue_state_tracking/dialogue_state_tracking_sgd.py (hunk 1 above)
Comment (VahidooX) [reply examples/nlp/dialogue_state_tracking/dialogue_state_tracking_sgd.py:158]: Done

Comment (VahidooX) [reply examples/nlp/dialogue_state_tracking/dialogue_state_tracking_sgd.py:158]: Done

File: examples/nlp/dialogue_state_tracking/dialogue_state_tracking_sgd.py (hunk 2 above)
Comment (VahidooX) [reply examples/nlp/dialogue_state_tracking/dialogue_state_tracking_sgd.py:165]: Done

File: nemo/collections/nlp/data/datasets/sgd_dataset/prediction_utils.py (hunk 4 above)
Comment (VahidooX) [reply nemo/collections/nlp/data/datasets/sgd_dataset/prediction_utils.py:33]: Done

File: examples/nlp/dialogue_state_tracking/dialogue_state_tracking_sgd.py (hunk 3 above)
Comment (VahidooX) [reply examples/nlp/dialogue_state_tracking/dialogue_state_tracking_sgd.py:224]: Done

File: nemo/collections/nlp/data/datasets/sgd_dataset/prediction_utils.py (hunk 5 above)
Comment (VahidooX) [reply nemo/collections/nlp/data/datasets/sgd_dataset/prediction_utils.py]: done

**Pull request merged**